feat(conv): blockfi parser

stevenmirabito opened the pull request:
Implements a conversion tool parser for [BlockFi](https://blockfi.com).

Same caveats as #31 and #69 apply - I've only used BlockFi for their earn product and not for lending, so this may not cover all transaction types. Unlike Nexo and Celsius, however, BlockFi does support buying/selling their supported symbols, and they provide a separate export for trade transactions. ~Unfortunately, I do not have an example export CSV to reference to implement a parser; when one becomes available, I think it would make sense to implement this like Binance and have a separate `TYPE_EXCHANGE` parser in addition to the `TYPE_WALLET` parser implemented in this PR.~

Sample input CSVs:

###### blockfi_wallet_sample.csv
```
Cryptocurrency,Amount,Transaction Type,Confirmed At
USDC,-1000,Withdrawal,1/16/21 9:48
BTC,0.00737669,Bonus Payment,1/15/21 23:59
USDC,7.025507246,Interest Payment,12/31/20 23:59
USDC,5.409836064,Interest Payment,12/1/20 4:59
USDC,10,Deposit,
BTC,0.0625,Trade,11/15/20 13:34
GUSD,-1000,Trade,11/15/20 13:34
USDC,1000,Deposit,11/6/20 16:21
GUSD,1000,Wire Deposit,11/1/20 9:30
```

###### blockfi_trades_sample.csv
```
Trade ID,Date,Buy Quantity,Buy Currency,Sold Quantity,Sold Currency,Rate Amount,Rate Currency,Type
63ff4494-135e-4f92-b8ec-6694c037ebf7,1/15/20 13:34,0.0625,btc,1000,gusd,16000,gusd,One Time
```

Expected output:
```
$ bittytax_conv -s blockfi_wallet_sample.csv blockfi_trades_sample.csv -o blockfi_out.xlsx
file: blockfi_wallet_sample.csv matched as "BlockFi"
row[6] ['USDC', '10', 'Deposit', '']
WARNING Skipping unconfirmed transaction, use the [-uc] option to include it
row[7] ['BTC', '0.0625', 'Trade', '11/15/20 13:34']
WARNING Skipping trade transaction, make sure you also import your BlockFi trading report
row[8] ['GUSD', '-1000', 'Trade', '11/15/20 13:34']
WARNING Skipping trade transaction, make sure you also import your BlockFi trading report
file: blockfi_trades_sample.csv matched as "BlockFi T"
gusd
output EXCEL file created: blockfi_out.xlsx
```

<img width="1025" alt="Screen Shot 2021-02-21 at 6 38 10 PM" src="https://user-images.githubusercontent.com/704880/108650473-80bf9680-7474-11eb-99d7-d9afc30d2eac.png">
<img width="977" alt="Screen Shot 2021-02-21 at 6 38 27 PM" src="https://user-images.githubusercontent.com/704880/108650477-81f0c380-7474-11eb-9e6f-a3790e3b5f88.png">

nanonano commented:
Hi @stevenmirabito, thanks for this parser!
Is there a way the Withdrawal Fee and the Withdrawal can be correlated, do you have an example row for it, do the timestamps match?

stevenmirabito commented:
[I asked on Reddit](https://www.reddit.com/r/blockfi/comments/lp3q10/blockfi_transaction_history_csv) to see if anyone could provide an example for the withdrawal fee, as well as the trading/lending products. I did get an example of the trading export, so I've updated the parser to support that as well and updated the example.

nanonano commented:
Added BlockFi parser as part of 71563dc.

stevenmirabito commented:
Looks great! Glad I could be of assistance.
